src/App.js

```diff
@@ -80,32 +81,39 @@ const App = (props) => {
         } catch (e) {
             chrome.hideGlobalFilter();
         }
+
+        setHasSystemsReady(true);
     }
 
     useEffect(() => {
         initChrome();
         // eslint-disable-next-line react-hooks/exhaustive-deps
     }, []);
 
+    if (!arePermissionsReady || !hasSystemsReady) {
+        return <PageLoading />;
+    }
+
+    if (hasSystemsReady && !hasSystems) {
```

Comment:
hasSystemsReady here is redundant because it will be always true if interpreter reaches this code.

Reply:
^ could be a reason why I find Sebastian's pick on the multiple return's making sense. I would rather nest conditions under one return. But both options are OK for me.

Reply:
Take a look at the comments below. 

We should separate this better, in separate components. The `hasSystems` and the `permissions` don't necessarily belong into the `App` component. If the `ZeroState` (or wrapper component), would contain the `hasSystems` state and render the children or the loading state depending on that, and we had a `PermissionsProvider` component doing something similar, everything would be neatly separated, there wouldn't be multiple returns and no complexity.